### PR TITLE
Content Negotiation for Express.js HTTP Thing

### PR DESCRIPTION
**Pull request by SergioCasCeb**
Continuing https://github.com/eclipse-thingweb/test-things/pull/15
- Modified the HTTP Thing to include headers for content negotiation (Text, JSON, and CBOR)
- Reflected such information on the generated TD
- Created a simple client to test the functionality of the Thing

### CODE REVIEW

**Review by egekorkan — CHANGES_REQUESTED** [review]
1. We should reduce content types to cbor and json since more means more forms in each affordance.
2. We should have another Thing code for content negotiation since we are complicating the simple protocol Thing. We should probably do this for all Things we have. Each protocol has more complex/edge cases in addition to simple ones. Trying to put all mechanisms into one Thing worsens the experience for users who want to test simple mechnanisms.

**Comment by egekorkan** [comment]
The comment and changes above will apply to CoAP as well.

File: things/calculator/http/client/README.md [hunk 1]
```diff
@@ -1,13 +1,23 @@
-# HTTP Test Client
+# HTTP Test Clients
 
-This client was created with the sole purpose of testing the main functionality of the HTTP Calculator Things such as the content negotiation.
+These clients were created with the sole purpose of testing the main functionality of the HTTP Calculator Things such as the content negotiation.
 
 **This folder and files are not final and might be subject to changes or might be completely removed in the future.**
 
-## Enpoints
+## Enpoints Simple Calculator Thing
```
Comment (danielpeintner) [inline things/calculator/http/client/README.md:7]:
```suggestion
## Endpoints Simple Calculator Thing
```

I think I noted that somewhere else already..

File: things/calculator/http/client/README.md [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
-- Subtract: /http-express-calculator/actions/subtract
\ No newline at end of file
+- Full TD: /http-express-calculator-simple
+- Result: /http-express-calculator-simple/properties/result
+- LastChange: /http-express-calculator-simple/properties/lastChange
+- Add: /http-express-calculator-simple/actions/add
+- Subtract: /http-express-calculator-simple/actions/subtract
+- Update: /http-express-calculator-simple/events/update
+
+## Enpoints Content Negotiation Calculator Thing
```
Comment (danielpeintner) [inline things/calculator/http/client/README.md:16]:
```suggestion
## Endpoints Content Negotiation Calculator Thing
```

**Review by egekorkan — CHANGES_REQUESTED** [review]
An initial review. I will test locally later as well.

File: things/calculator/http/client-tests/content-negotiation-http-client.js [hunk 3]
```diff
@@ -0,0 +1,319 @@
+/**
+ * @file The `content-negotiation-http-client.js` file acts a client for the content-negotiation-calculator.js.
```
Comment (danielpeintner) [inline things/calculator/http/client-tests/content-negotiation-http-client.js:2]:
```suggestion
 * @file The `content-negotiation-http-client.js` file acts as a client for the content-negotiation-calculator.js.
```

**Comment by egekorkan** [comment]
Last fixes:
- Method not allowed should be added
- JSON Schema based validation should be used instead of type casting. Currently, sending `"1"` is the same as sending `1`.
- date-time should be used for last change and it should be formatted to match one of the version noted at https://json-schema.org/understanding-json-schema/reference/string#dates-and-times
- SSE endpoints for property observation does not work. For now, let's do observation with another endpoint (/result/observe). We will come to this when it is clarified within the spec

**Comment by egekorkan** [comment]
Merging. All tests pass now :)
